agent/src/test/java/com/teamscale/jacoco/agent/AgentHttpServerTest.java

```diff
@@ -0,0 +1,104 @@
+package com.teamscale.jacoco.agent;
+
+import com.teamscale.client.TeamscaleServer;
+import com.teamscale.jacoco.agent.options.AgentOptionParseException;
+import com.teamscale.jacoco.agent.options.AgentOptions;
+import com.teamscale.jacoco.agent.options.TestAgentOptionsBuilder;
+import com.teamscale.jacoco.agent.upload.UploaderException;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AgentHttpServerTest {
+
+	private Agent agent;
+	private final Integer httpServerPort = 8081;
+	private final String defaultCommitMessage = "Some Message";
+	private final String defaultPartition = "Some Partition";
+
+	/** Starts the http server to control the agent */
+	@BeforeEach
+	public void setup() throws AgentOptionParseException, UploaderException {
```

Comment:
General advice for tests. Don't bother with "precise" `throws` clauses. Just use `throws Exception` anywhere, as there is no-one handling the different types of exception specially. Just the JUnit runner doing the same for all exceptions: failing the tests.